actor/builtin/miner/miner_test.go

```diff
@@ -696,6 +699,116 @@ func TestMinerSubmitPoStPowerUpdates(t *testing.T) {
+		minerState.ProvingPeriodEnd = types.NewBlockHeight(ProvingPeriodDuration(types.OneKiBSectorSize))
+		minerState.SectorCommitments = NewSectorSet()
+		minerState.SectorCommitments.Add(1, types.Commitments{CommR: commR1})
+		minerState.SectorCommitments.Add(2, types.Commitments{CommR: commR2})
+		minerState.SectorCommitments.Add(3, types.Commitments{CommR: commR3})
+
+		// The 3 sector is not in the proving set, so its CommR should not appear in the VerifyPoSt request
+		minerState.ProvingSet = types.NewIntSet(1, 2)
+		vmctx := th.NewFakeVMContext(message, minerState)
+
+		verifier := &th.FakeVerifier{Valid: true}
+		vmctx.TestVerifier = verifier
+
+		miner := Actor{Bootstrap: false}
+
+		testProof := []types.PoStProof{th.MakeRandomPoStProofForTest()}
+		_, err := miner.SubmitPoSt(vmctx, testProof, types.NewIntSet())
+		require.NoError(t, err)
+
+		require.NotNil(t, verifier.PoStRequest)
+		assert.Equal(t, types.OneKiBSectorSize, verifier.PoStRequest.SectorSize)
+
+		seed := types.PoStChallengeSeed{}
+		copy(seed[:], vmctx.TestRandomness)
+
+		sortedRs := proofs.NewSortedCommRs(commR1, commR2)
+
+		assert.Equal(t, seed, verifier.PoStRequest.ChallengeSeed)
+		assert.Equal(t, 0, len(verifier.PoStRequest.Faults))
+		assert.Equal(t, 1, len(verifier.PoStRequest.Proofs))
+		assert.Equal(t, testProof, verifier.PoStRequest.Proofs)
+		assert.Equal(t, 2, len(verifier.PoStRequest.SortedCommRs.Values()))
+		assert.Equal(t, sortedRs.Values()[0], verifier.PoStRequest.SortedCommRs.Values()[0])
+		assert.Equal(t, sortedRs.Values()[1], verifier.PoStRequest.SortedCommRs.Values()[1])
+	})
+
+	t.Run("Faults if proving set commitment is missing from sector commitments", func(t *testing.T) {
+		minerState := *NewState(address.TestAddress, address.TestAddress, peer.ID(""), types.OneKiBSectorSize)
+		minerState.ProvingPeriodEnd = types.NewBlockHeight(ProvingPeriodDuration(types.OneKiBSectorSize))
+		minerState.SectorCommitments = NewSectorSet()
+
+		minerState.ProvingSet = types.NewIntSet(4)
+		vmctx := th.NewFakeVMContext(message, minerState)
+
+		miner := Actor{Bootstrap: false}
+
+		testProof := []types.PoStProof{th.MakeRandomPoStProofForTest()}
+		code, err := miner.SubmitPoSt(vmctx, testProof, types.NewIntSet())
+		require.Error(t, err)
+		assert.Equal(t, "miner ProvingSet sector id 4 missing in SectorCommitments", err.Error())
+		assert.True(t, errors.IsFault(err))
+		assert.Equal(t, uint8(1), code)
+	})
+
+	t.Run("Reverts if verification errors", func(t *testing.T) {
+		minerState := *NewState(address.TestAddress, address.TestAddress, peer.ID(""), types.OneKiBSectorSize)
+		minerState.ProvingPeriodEnd = types.NewBlockHeight(ProvingPeriodDuration(types.OneKiBSectorSize))
+		minerState.SectorCommitments = NewSectorSet()
+		minerState.SectorCommitments.Add(1, types.Commitments{CommR: commR1})
+
+		minerState.ProvingSet = types.NewIntSet(1)
+		vmctx := th.NewFakeVMContext(message, minerState)
+
+		verifier := &th.FakeVerifier{Err: xerrors.New("verifier error")}
```

Comment:
This looks unintentional

Reply:
It wasn't. It's a pattern that's been in place when "errors" and "github.com/filecoin-project/go-filecoin/vm/errors" are both in use. I can swap the namespaces to "errors" and "vmerrors" though.